src/components/canrawview/canrawview_p.h

```diff
@@ -90,17 +121,41 @@ class CanRawViewPrivate : public QWidget {
     CanRawView* q_ptr;
     int prevIndex = 0;
     int rowID = 0;
+    QStringList columnsOrder;
+
+    void writeColumnsOrder(QJsonObject& json) const
+    {
+        uint ii = 0;
+        QJsonArray columnList;
+        for (auto iter : columnsOrder) {
+            if (ui->tv->isColumnHidden(ii++) == false) {
```

Comment:
Can we go with `const auto& column` instead of a `auto iter`?

I would remove `ii++` from here and put it after `columnList.append(...` as `++index;`.

I would add `assert(nullptr != ui);` and `assert(nullptr != ui->tv);`.